core/cache/application.go

```diff
@@ -28,18 +33,35 @@ type Application struct {
 	configHash string
 }
 
-func (m *Application) setDetails(details ApplicationChange) {
-	m.mu.Lock()
-	defer m.mu.Unlock()
-	m.details = details
+func (a *Application) WatchFields(comparitors ...func(*ApplicationDelta) bool) *ApplicationFieldWatcher {
```

Comment:
my initial reaction was that it was strange to pass in a list of functions, rather than passing a list of strings/constants/attributes.
The existing Watch implementations have supported a 'filter' function, though what that filter could be was obviously quite a bit different. But functionally that is exactly what you are doing here.
WatchFiltered(comparitors)
The comparitors themselves happen to care about specific fields, and that is what you are automatically generating. But it is just a filter function. 
Maybe its just the naming. If you called this WatchWithFilters() and passed in field filters, vs WatchFields but you don't pass in 'fields'.